service/config_provider.go

```diff
@@ -163,11 +187,20 @@ func (cm *configProvider) Shutdown(ctx context.Context) error {
 	return multierr.Combine(cm.closeIfNeeded(ctx), mergeShutdown(ctx, cm.configMapProviders))
 }
 
-func mergeRetrieve(ctx context.Context, onChange func(*configmapprovider.ChangeEvent), providers []configmapprovider.Provider) (configmapprovider.Retrieved, error) {
+func mergeRetrieve(ctx context.Context, onChange func(*configmapprovider.ChangeEvent), selectors []string, providers map[string]configmapprovider.Provider) (configmapprovider.Retrieved, error) {
 	var retrs []configmapprovider.Retrieved
 	retCfgMap := config.NewMap()
-	for _, p := range providers {
-		retr, err := p.Retrieve(ctx, onChange)
+	for _, selector := range selectors {
+		// For backwards compatibility, empty url scheme means "file".
+		scheme := "file"
+		if idx := strings.Index(selector, ":"); idx != -1 {
+			scheme = selector[:idx]
+		}
+		p, ok := providers[scheme]
+		if !ok {
+			return nil, configmapprovider.ErrSchemeIsNotSupported
```

Comment:
It would be useful to include the name of the unsupported scheme in the error message.

Reply:
Done.